drivers/ethernet/eth_lan865x.c

```diff
@@ -0,0 +1,551 @@
+		LOG_ERR("Failed to configure reset GPIO, %d", ret);
+		return ret;
+	}
+
+	/* Perform (GPIO based) HW reset */
+	/* assert RESET_N low for 10 µs (5 µs min) */
+	gpio_pin_set_dt(&cfg->reset, 1);
+	k_busy_wait(10U);
+	/* deassert - end of reset indicated by IRQ_N low  */
+	gpio_pin_set_dt(&cfg->reset, 0);
+
+	return lan865x_wait_for_reset(dev);
+}
+
+static int lan865x_port_send(const struct device *dev, struct net_pkt *pkt)
+{
+	struct lan865x_data *ctx = dev->data;
+	struct oa_tc6 *tc6 = ctx->tc6;
+	int ret;
+
+	k_sem_take(&ctx->tx_rx_sem, K_FOREVER);
+	ret = oa_tc6_send_chunks(tc6, pkt);
+	k_sem_give(&ctx->tx_rx_sem);
+	if (ret < 0) {
+		LOG_ERR("TX transmission error, %d", ret);
+		eth_stats_update_errors_tx(net_pkt_iface(pkt));
+		return ret;
+	}
+
+	return 0;
+}
+
+static const struct ethernet_api lan865x_api_func = {
+	.iface_api.init = lan865x_iface_init,
+	.get_capabilities = lan865x_port_get_capabilities,
+	.set_config = lan865x_set_config,
+	.send = lan865x_port_send,
+};
+
+#define LAN865X_DEFINE(inst)                                                                      \
+	static const struct lan865x_config lan865x_config_##inst = {                     \
+		.spi = SPI_DT_SPEC_INST_GET(inst, SPI_WORD_SET(8), 0),                             \
+		.interrupt = GPIO_DT_SPEC_INST_GET(inst, int_gpios),                               \
+		.reset = GPIO_DT_SPEC_INST_GET(inst, rst_gpios),                               \
+		.timeout = CONFIG_ETH_LAN865X_TIMEOUT,                                            \
+		.plca_node_id = DT_INST_PROP(inst, plca_node_id), \
+		.plca_node_count = DT_INST_PROP(inst, plca_node_count), \
+		.plca_burst_count = DT_INST_PROP(inst, plca_burst_count), \
+		.plca_burst_timer = DT_INST_PROP(inst, plca_burst_timer), \
+		.plca_to_timer = DT_INST_PROP(inst, plca_to_timer), \
+		.plca_enable = DT_INST_PROP(inst, plca_enable), \
+	};                                                                                         \
+	\
+	struct oa_tc6 oa_tc6_##inst = {                                                           \
+		.cps = 64, \
+		.protected = 0, \
+		.spi = &lan865x_config_##inst.spi \
+	}; \
+	static struct lan865x_data lan865x_data_##inst = {                                   \
+		.mac_address = DT_INST_PROP(inst, local_mac_address),                              \
+		.tx_rx_sem =                                                                       \
+			Z_SEM_INITIALIZER((lan865x_data_##inst).tx_rx_sem, 1, UINT_MAX),   \
+		.int_sem = Z_SEM_INITIALIZER((lan865x_data_##inst).int_sem, 0, UINT_MAX),  \
+		.tc6 = &oa_tc6_##inst \
+	};                                                                                         \
+                                                                                                   \
+	ETH_NET_DEVICE_DT_INST_DEFINE(inst, lan865x_init, NULL, &lan865x_data_##inst, \
+				      &lan865x_config_##inst, CONFIG_ETH_INIT_PRIORITY,       \
+				      &lan865x_api_func, NET_ETH_MTU);
```

Comment:
Minor nit but could you adjust the \\ at the end of the line to be in the same line, it looks hurts to look at it.